packages/contracts/hardhat.config.js

```diff
@@ -1,16 +1,79 @@
+const { task } = require("hardhat/config");
+
+require("@nomiclabs/hardhat-etherscan");
 require('@nomiclabs/hardhat-ethers');
 require('@nomiclabs/hardhat-waffle');
 require('hardhat-deploy');
 require('hardhat-deploy-ethers');
 
+real_accounts = undefined;
+if(process.env.DEPLOYER_KEY && process.env.OWNER_KEY) {
+  real_accounts = [process.env.DEPLOYER_KEY, process.env.OWNER_KEY];
+}
+const gatewayurl = "https://offchain-resolver-example.uc.r.appspot.com/{sender}/{data}.json"
 /**
  * @type import('hardhat/config').HardhatUserConfig
  */
+
+// Usage: INFURA_ID=$INFURA_ID ETHERSCAN_API_KEY=$ETHERSCAN_API_KEY npx hardhat  --network $NETWORK etherscanverify --resolver $CONTRACT_ADDRESS --signer $SIGNER_ADDRESS"
+task("etherscanverify", "Verifies contracts")
```

Comment:
yes, I tried but the default one doesn't let me pass gatewayurl from the network config.

Reply:
You don't have to; it verifies using the actual deployment parameters.

Reply:
I get `Error HH305: Unrecognized param --apiurl` when I pass `npx hardhat --network mainnet etherscan-verify  --api-key $ETHERSCAN_API --apiurl  https://mainnet.infura.io/v3/INFURA_ID` and get `Error HH110: Invalid JSON-RPC response received: invalid project id` without it

Reply:
You don't need `apiurl`; it will use the RPC URL from the hardhat config for the network you selected. You need to specify `INFURA_ID` on the command line so it's substituted into the config.